sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -556,6 +556,8 @@ static az_result ready(az_event_policy* me, az_event event)
       _az_RETURN_IF_FAILED(az_rpc_client_get_response_topic(
           this_policy->_internal.rpc_client,
           event_data->rpc_server_client_id,
+          _az_span_is_valid(event_data->command_name, 1, 0) ? event_data->command_name
```

Comment:
Is there a case where the span is invalid on purpose?
If not, I think the checks should be done as preconditions rather than product code.

If, for example, the span _was_ valid but got corrupted (so that the size is now negative for example), this code will ignore the corruption and pass an empty span.
Instead, we should only accept _valid_ spans (including empty) and perform the validation within the codec function.

Reply:
yep! The application has the option of passing in the command name for this specific invocation or passing AZ_SPAN_EMPTY and using the one that's on the client codec object.